src/soundio/soundmanager.cpp

```diff
@@ -559,8 +559,8 @@ SoundDeviceStatus SoundManager::setConfig(const SoundManagerConfig& config) {
     // certain parts of mixxx rely on this being here, for the time being, just
     // letting those be -- bkgood
     // Do this first so vinyl control gets the right samplerate -- Owen W.
-    m_pConfig->set(ConfigKey("[Soundcard]","Samplerate"),
-                   ConfigValue(static_cast<int>(m_config.getSampleRate())));
+    m_pConfig->set(ConfigKey("[Soundcard]", "Samplerate"),
+            ConfigValue(static_cast<int>(m_config.getSampleRate().value())));
```

Comment:
```
/home/jan/Projects/mixxx/src/soundio/soundmanager.cpp: In member function ‘SoundDeviceStatus SoundManager::setConfig(const SoundManagerConfig&)’:
/home/jan/Projects/mixxx/src/soundio/soundmanager.cpp:563:57: error: call of overloaded ‘ConfigValue(mixxx::audio::SampleRate::value_t)’ is ambiguous
  563 |             ConfigValue(m_config.getSampleRate().value()));
      |                                                         ^
In file included from /home/jan/Projects/mixxx/src/preferences/usersettings.h:6,
                 from /home/jan/Projects/mixxx/src/controllers/midi/midimessage.h:10,
                 from /home/jan/Projects/mixxx/src/control/controlbehavior.h:6,
                 from /home/jan/Projects/mixxx/src/control/control.h:9,
                 from /home/jan/Projects/mixxx/src/control/pollingcontrolproxy.h:6,
                 from /home/jan/Projects/mixxx/src/soundio/soundmanager.h:11,
                 from /home/jan/Projects/mixxx/src/soundio/soundmanager.cpp:1:
/home/jan/Projects/mixxx/src/preferences/configobject.h:83:14: note: candidate: ‘ConfigValue::ConfigValue(double)’
   83 |     explicit ConfigValue(double value);
      |              ^~~~~~~~~~~
/home/jan/Projects/mixxx/src/preferences/configobject.h:82:14: note: candidate: ‘ConfigValue::ConfigValue(int)’
   82 |     explicit ConfigValue(int value);
      |              ^~~~~~~~~~~
/home/jan/Projects/mixxx/src/preferences/configobject.h:73:7: note: candidate: ‘ConfigValue::ConfigValue(const ConfigValue&)’
   73 | class ConfigValue {
      |       ^~~~~~~~~~~
```

Reply:
okay, Ignore for now. I'll look into it some other time.

Reply:
I've looked into it. The problem is trivial on the surface (uint32_t can't be promoted to int nor double, thus it would be converted and neither conversion ranks better than the other). The problem gets harder when I looked into fixing it. `configobject.h` is unnecessarily complex with lots of duplicated code due to a lack of separation of concerns (`ConfigValue` vs `ConfigObject::setValue`, which type/function is supposed to do the conversion?). I may do some refactoring in the future.